### PR TITLE
Update Travis build environment to Xcode 12

### PR DESCRIPTION
**Pull request by mtgto**
Update Travis build environment to Xcode 12, and fix compiler/test errors while updating.
https://docs.travis-ci.com/user/reference/osx/#xcode-version

- Set platform to iPhone 12 (from iPhone 7).
  - [travis.yml choose latest iOS](https://github.com/abbeycode/UnrarKit/blob/2.9/.travis.yml#L18).
  - Current latest iOS is iOS 14.2. It can not be used in iPhone 7 at Travis CI. For more information: https://travis-ci.org/github/mtgto/UnrarKit/jobs/752360232
- Use Carthage workaround about build error with Xcode 12.
  - https://github.com/Carthage/Carthage/blob/master/Documentation/Xcode12Workaround.md
- Fix URKxxx macros to force semicolon to resolve compile warnings
- Update Tests/TestData/bin/rar to latest 64bit binary (`RAR 6.00 for macOS (64 bit)` https://www.rarlab.com /download.htm )
  - macOS 10.15+ can't execute 32bit executable binary
- Set locale of test to `English (US)`.
  - `NSProgress.localizedDescription` is changed by user locale.

### CODE REVIEW

File: Tests/URKArchiveTestCase.m [hunk 1]
```diff
@@ -352,11 +352,11 @@ - (NSURL *)archiveWithFiles:(NSArray *)fileURLs name:(NSString *)archiveName arg
 {
     NSURL *textFile = [self randomTextFileOfLength:100000];
     
-    // Generate multi-volume archive, with parts no larger than 20 KB in size
+    // Generate multi-volume RAR 4.0 archive, with parts no larger than 20 KB in size
     NSString *commandOutputString = nil;
     [self archiveWithFiles:@[textFile]
                       name:baseName
-                 arguments:@[@"-v20k", @"-vn"]
+                 arguments:@[@"-v20k", @"-ma4", @"-vn"]
```
Comment (mtgto) [inline Tests/URKArchiveTestCase.m:359]: This change need by updating rar executable binary.

`-vn` option can not use in RAR 5.0 format.
`-ma4` means rar generates RAR 4.0 archive format.

> Creating old style volume names in archive.r00, archive.r01 format is not supported.
https://www.winrar-france.fr/winrar_instructions_for_use/source/html/HELPRAR5Format.htm

**Comment by mtgto** [comment]
Validation of Cocoapods failed due to length of log.

> The job exceeded the maximum log length, and has been terminated.

Do I use `pod lib lint` without `--verbose` option...?

**Comment by abbeycode** [comment]
@mtgto Have you looked at the `v2.10` branch? Those builds already target Xcode 12 in Travis, and it's where active development is happening. Regarding the log length, that is currently affecting the `v2.10` branch as well. I've handled it in the past by configuring the log level (the `before_script` section in the .travis.yml file), so I'll have to look into why that's not working now. Thanks for pointing it out.

**Comment by abbeycode** [comment]
@mtgto The `v2.10` branch was failing for an unrelated reason, and I've fixed it. I'm going to close this PR, assuming that branch will meet your needs. Please feel free to reopen if there are other changes you feel make sense, but retarget to `v2.10` instead of master.

**Comment by mtgto** [comment]
> Have you looked at the v2.10 branch? Those builds already target Xcode 12 in Travis, and it's where active development is happening. Regarding the log length, that is currently affecting the v2.10 branch as well.

I don't know before I create this pull request.
I hope you think about choosing default branch as `v2.10' if you don't want to send a pull request to master branch 😃 

This pull request contains a bit changes, such as replacing old rar, setting default time locale.
You can use my pull request code as you can If you want to. Thanks for reply 👍

**Comment by abbeycode** [comment]
Thanks for the default branch suggestion. I haven't done that before, but have done so now. I never intend for my version branches to stick around long, but they always seem to. That'll help avoid confusion.

Looking through your other changes, I was curious about the `carthage.sh` script you added. I don't see it referred to in tests or the Travis config - what was that for?

**Comment by mtgto** [comment]
> Looking through your other changes, I was curious about the carthage.sh script you added. I don't see it referred to in tests or the Travis config - what was that for?

I met a trouble CI failure with carthage and Xcode 12, which is you already fixed by same way of me.

https://travis-ci.org/github/mtgto/UnrarKit/jobs/752368996
https://github.com/abbeycode/UnrarKit/commit/5e05db844c6d4066b51e5f6c45e71e79081975f9

**Pull request closed without merging**